src/Identifier/LdapIdentifier.php

```diff
@@ -0,0 +1,240 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Authentication\Identifier;
+
+use Authentication\Identifier\Backend\Ldap;
+use Authentication\Identifier\Backend\LdapInterface;
+use Cake\Core\App;
+use Cake\Core\Exception\Exception;
+use Cake\Network\Exception\InternalErrorException;
+use Cake\ORM\Entity;
+use ErrorException;
+use InvalidArgumentException;
+use RuntimeException;
+
+/**
+ * LDAP Identifier
+ *
+ * Identifies authentication credentials using LDAP.
+ *
+ * ```
+ *  new LdapIdentifier([
+ *       'host' => 'ldap.example.com',
+ *       'bindDN' => function($username) {
+ *           return $username; //transform into a rdn or dn
+ *       },
+ *       'options' => [
+ *           LDAP_OPT_PROTOCOL_VERSION => 3
+ *       ]
+ *  ]);
+ * ```
+ *
+ * @link https://github.com/QueenCityCodeFactory/LDAP
+ */
+class LdapIdentifier extends AbstractIdentifier
+{
+
+    /**
+     * Default configuration
+     *
+     * @var array
+     */
+    protected $_defaultConfig = [
+        'ldap' => Ldap::class,
+        'fields' => [
+            'username' => 'username',
+            'password' => 'password'
+        ],
+        'port' => 389
+    ];
+
+    /**
+     * List of errors
+     *
+     * @var array
+     */
+    protected $_errors = [];
+
+    /**
+     * LDAP connection object
+     *
+     * @var \Authentication\Identifier\Backend\LdapInterface
+     */
+    protected $_ldap = null;
+
+    /**
+     * {@inheritDoc}
+     */
+    public function __construct(array $config = [])
+    {
+        parent::__construct($config);
+
+        $this->_checkLdapExtension();
+        $this->_checkLdapConfig();
+        $this->_buildLdapObject();
+    }
+
+    /**
+     * Checks if the php LDAP extension is loaded
+     *
+     * @throws \RuntimeException
+     * @return void
+     */
+    protected function _checkLdapExtension()
+    {
+        if (!extension_loaded('ldap')) {
```

Comment:
This check should be done in LDAP adapter. If someone wanted to use their own extension-independent adapter, this would fail.